### PR TITLE
fix(challenge): More robust test for "Learn how a Stack Works".

### LINKED ISSUE #13459: Learn How a Stack Works Insufficient Tests
#### Challenge Name

http://beta.freecodecamp.com/en/challenges/coding-interview-data-structure-questions/learn-how-a-stack-works

#### Issue Description

The challenge tests are insufficient and can be passed with incorrect code. The code shown below can pass the challenge but it doesn't use any of the JavaScript methods to complete the challenge.

#### Browser Information

* Browser Name, Version: Chrome
* Operating System: macOS
* Mobile, Desktop, or Tablet: Laptop 

#### Code

```js
var homeworkStack = ['BIO12','HIS80','MAT122','PSY44'];
// Only change code below this line
homeworkStack = ['BIO12', 'HIS80','MAT122', 'CS50'];
```

#### Screenshot

image

### PR DESCRIPTION
**Pull request by Greenheart**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #13459 

#### Description

Improve robustness of the tests by ensuring that the `homeworkStack` wasn't overwritten.

### CODE REVIEW

File: seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json [hunk 1]
```diff
@@ -63,7 +63,8 @@
       "tests": [
         "assert(homeworkStack.length === 4, 'message: <code>homeworkStack</code> should only contain 4 elements.');",
         "assert(homeworkStack[3] === 'CS50', 'message: The last element in <code>homeworkStack</code> should be <code>\"CS50\"</code>.');",
-        "assert(homeworkStack.indexOf('PSY44') === -1, 'message: <code>homeworkStack</code> should not contain <code>\"PSY44\"</code>.');"
+        "assert(homeworkStack.indexOf('PSY44') === -1, 'message: <code>homeworkStack</code> should not contain <code>\"PSY44\"</code>.');",
+        "assert(code.match(/=/g).length === 1 && /homeworkStack\\s*=\\s*\\[\"BIO12\"\\s*,\\s*\"HIS80\"\\s*,\\s*\"MAT122\"\\s*,\\s*\"PSY44\"\\]/.test(code)), 'message: The initial declaration of the <code>homeworkStack</code> should not be changed.'"
```
Comment (erictleung) [inline seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json]: You're missing a closing parens and semicolon :smile: 

```diff
- <code>homeworkStack</code> should not be changed.'"
+ <code>homeworkStack</code> should not be changed.');"
```

So it's creating a stray single quote in the tests

image

**Review by erictleung — CHANGES_REQUESTED** [review]
Close, just a small change so it renders correctly.

File: seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json (hunk 1 above)
Comment (Greenheart) [inline seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-data-structure-questions.json]: @erictleung Haha nice catch! :smile:

**Comment by Greenheart** [comment]
@erictleung Should be fixed now :wink:

**Review by erictleung — APPROVED** [review]
Sorry for the late QA! Been busy lately. Nonetheless, LGTM! 👍  Thanks for working on this!

**Comment by Greenheart** [comment]
@erictleung Haha no worries! I've been away for the last few weeks myself :smile:

**Comment by QuincyLarson** [comment]
@Greenheart awesome work on these. Thanks for all these contributions.

**Pull request merged**